algo/src/main/java/org/neo4j/graphalgo/impl/LabelPropagation.java

```diff
@@ -0,0 +1,74 @@
+package org.neo4j.graphalgo.impl;
+
+import com.carrotsearch.hppc.DoubleDoubleHashMap;
+import com.carrotsearch.hppc.DoubleDoubleMap;
+import com.carrotsearch.hppc.IntDoubleHashMap;
+import com.carrotsearch.hppc.IntDoubleMap;
+import com.carrotsearch.hppc.cursors.DoubleDoubleCursor;
+import org.neo4j.graphalgo.api.RelationshipConsumer;
+import org.neo4j.graphalgo.core.heavyweight.HeavyGraph;
+import org.neo4j.graphdb.Direction;
+
+import java.util.function.IntConsumer;
+
+public final class LabelPropagation implements IntConsumer, RelationshipConsumer {
+
+    private final HeavyGraph graph;
```

Comment:
It needs the NodeWeights and NodeProperty interfaces. Since we decided to first implement the algos on the HeavyGraph only, I had only this one implement the interfaces. Moving to `Graph` is certainly as TODO.